### ISSUE TITLE
Basic authentication throws unauthorized error

### ISSUE THREAD

**Issue by maxfrischknecht** [johannschopplich/nuxt-kql#5]
Hi! 

First thank you for taking the time to develop this module, I was really exited when I saw it!

However, I'm always running into the same error: `Error: 500 Failed to execute KQL query (/api/__kql/%24kqlOVgqL06EA2)`

<img width="710" alt="Bildschirmfoto 2023-02-04 um 12 55 06" src="https://user-images.githubusercontent.com/38426074/216782949-bf80f9c6-a16b-4357-8f56-5264f4a30ae5.png">

Kirby Setup:
- Kirby Plainkit ^3.8
- Kql ^2.0

Running locally with Docker (I also tried MAMP and on a remote Server, the error stays the same)

Nuxt Setup:
- Nuxt ^3.1.0
- Nuxt Kql ^0.25.5

Based on a fresh Nuxt3 install, I also tried your [kirby-nuxt-starterkit](https://github.com/johannschopplich/kirby-nuxt-starterkit) the error stays the same.

I also disabled `'auth' => false` in the Kirby KQL Plugin, this lets me consume the data with a regular `useAsyncData` function, however, with the plugin the error stays the same. 

Any ideas what the problem could be?
Thx for any help!

**Comment by johannschopplich** [comment]
Hi Max!

Thank you for raising the issue. Multiple factors turned out to be the reason for the error you encountered:

- The docs should state more clearely, that the [kirby-headless-starter](https://github.com/johannschopplich/kirby-headless-starter) is the way to for Nuxt KQL projects. Primarily for its token-based authentication method. I've updated the [getting started guide](https://nuxt-kql.jhnn.dev/guide/getting-started.html#step-3-set-up-the-environment) to reflect that recommendation.
- Although I recommend the Kirby Headless Starter, `nuxt-kql` should work as well with basic authentication in a fresh project like the plainkit + kql plugin you mentioned. 
  - Fixed a bug in https://github.com/johannschopplich/nuxt-kql/commit/0f6b9ad8ae7865a4d852c1d1ee7963013d0ae7b9 when building the `Authentication` header.
  - Side note: The internal Nuxt API proxy will always return status code 500. The reason for the reject itself was 401 unauthorized tho.

In a nutshell: I recommend the [kirby-headless-starter](https://github.com/johannschopplich/kirby-headless-starter) with its token-based authentication and pre-configured KQL settings. If you intend to stick with a fresh plainkit, please update to v0.25.6. Thanks!

**Comment by maxfrischknecht** [comment]
Hi Johann!

Thx for getting back & for the docs update. I can confirm that this works!

One thing that is maybe worth adding to the docs is that the `KIRBY_BASE_URL` shouldn't contain `api/query` at the end.
So `KIRBY_BASE_URL=http://localhost/` works, while `KIRBY_BASE_URL=http://localhost/api/query` doesn't because your plugin adds it I guess. Which is nice, but I had to test both versions. 

Thx again! 👏

**Comment by johannschopplich** [comment]
Glad to hear it's working. I've updated the docs to make clear that the base URL intends to contain the origin of the Kirby instance only, without any path.

If you have any further questions, feel free to contact me. In any case: Have fun with `nuxt-kql`!
